app/src/main/java/be/chvp/nanoledger/data/parser/TransactionParser.kt

```diff
@@ -38,21 +38,34 @@ fun extractTransactions(lines: List<String>): List<Transaction> {
     return result
 }
 
-val commentRegex = Regex(";.*$")
+val commentRegex = Regex("[ \\t]*;.*$")
 val postingSplitRegex = Regex("[ \\t]{2,}")
 
 fun extractPosting(line: String): Posting? {
-    val stripped = line.trim().replace(commentRegex, "")
-    if (stripped.length == 0) {
-        return null
+    // the three components of a posting
+    var account: String? = null
+    var amount: Amount? = null
+    var note: String? = null
+
+    // check if we have a note in the posting
+    val commentMatch = commentRegex.find(line)
```

Comment:
This should probably be done on `line.trim()`. You can actually see it in your screenshot that the first comment/note is somewhat offset from the postings. I think it'd look better if it was nicely aligned.

Reply:
One thing we would lose when doing it this way is the amount of space between the amount and the comment/note. So maybe it's better to fix that misalignment at the UI layer by doing a trim there. WDYT?

Reply:
Done with the last change, in the parser we store the spaces but in the UI we trim the note so it looks nice